File: php/euler3.php
#!/usr/bin/php
<?php

$number = 600851475143;
$divisor = 2;
while ($number > 1) {
	if ($number % $divisor == 0)
	{
		$number /= $divisor;
		$divisor--;
	}
	$divisor++;
}
echo $divisor . PHP_EOL;

?>

File: c++/euler3.cpp
#include <iostream>

int main()
{
	long long int number = 600851475143;
	int divisor = 2;
	while (number > 1) {
		if (0 == (number % divisor)) {
			number /= divisor;
			divisor--;
		}
		divisor++;
	}
	std::cout << divisor << std::endl;
	return 0;
}

